numpyro/infer/elbo.py

```diff
@@ -135,29 +158,65 @@ def single_particle_elbo(rng_key):
                 if site["type"] == "mutable"
             }
             params.update(mutable_params)
-            seeded_model = replay(seeded_model, guide_trace)
-            model_log_density, model_trace = log_density(
-                seeded_model, args, kwargs, params
-            )
-            check_model_guide_match(model_trace, guide_trace)
-            _validate_model(model_trace, plate_warning="loose")
-            mutable_params.update(
-                {
+            if self.multi_sample_guide:
+                plates = {
                     name: site["value"]
-                    for name, site in model_trace.items()
-                    if site["type"] == "mutable"
+                    for name, site in guide_trace.items()
+                    if site["type"] == "plate"
                 }
-            )
+
+                def get_model_density(key, latent):
+                    with seed(rng_seed=key), substitute(data={**latent, **plates}):
+                        model_log_density, _ = log_density(model, args, kwargs, params)
+                    return model_log_density
+
+                num_guide_samples = None
+                for name, site in guide_trace.items():
+                    if site["type"] == "sample":
+                        num_guide_samples = site["value"].shape[0]
+                        break
+                if num_guide_samples is None:
+                    raise ValueError("guide is missing `sample` sites.")
+                seeds = random.split(model_seed, num_guide_samples)
+                latents = {
+                    name: site["value"]
+                    for name, site in guide_trace.items()
+                    if (site["type"] == "sample" and site["value"].size > 0)
+                    or (site["type"] == "deterministic")
+                }
+                model_log_density = vmap(get_model_density)(seeds, latents)
+                assert model_log_density.ndim == 1
+                model_log_density = model_log_density.sum(0)
+                # log p(z) - log q(z)
+                elbo_particle = (model_log_density - guide_log_density) / seeds.shape[0]
+            else:
+                seeded_model = seed(model, model_seed)
+                replay_model = replay(seeded_model, guide_trace)
+                model_log_density, model_trace = log_density(
+                    replay_model, args, kwargs, params
+                )
+                check_model_guide_match(model_trace, guide_trace)
```

Comment:
so we lose these two checks if `multi_sample_guide==True`?

Reply:
we can check the later one. The first one is tricky.